### PR TITLE
Transcripts completeness

### PR DESCRIPTION
**Pull request by northwestwitch**
### This PR adds | fixes:
- Returns coverage completeness also for gene transcripts coverage queries

### How to test:
- Run a query for coverage over the transcripts of one or more gene for a sample in the database
- Provide also completeness thresholds

### Expected outcome:
- [x] Completeness stats should be returned in the response

### Review:
- [ ] Code approved by
- [x] Tests executed by CR

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by northwestwitch** [comment]
Local test:

image


Returns:

```
[
  {
    "chromosome": "1",
    "completeness": [
      [
        10,
        0.9891189891189891
      ],
      [
        20,
        0.6167076167076168
      ],
      [
        30,
        0.03159003159003159
      ]
    ],
    "ensembl_gene_id": "ENSG00000143632",
    "end": 229569845,
    "hgnc_id": 129,
    "hgnc_symbol": "ACTA1",
    "interval_id": null,
    "mean_coverage": 21.42892242892243,
    "start": 229566992
  },
  {
    "chromosome": "18",
    "completeness": [
      [
        10,
        0.9932827289097805
      ],
      [
        20,
        0.6964961261610477
      ],
      [
        30,
        0.1489649318214158
      ]
    ],
    "ensembl_gene_id": "ENSG00000101680",
    "end": 7117813,
    "hgnc_id": 6481,
    "hgnc_symbol": "LAMA1",
    "interval_id": null,
    "mean_coverage": 24.366699235041196,
    "start": 6941743
  },
  {
    "chromosome": "7",
    "completeness": [
      [
        10,
        0.9990151164503166
      ],
      [
        20,
        0.8889579958342976
      ],
      [
        30,
        0.35653967936714986
      ]
    ],
    "ensembl_gene_id": "ENSG00000128513",
    "end": 124570037,
    "hgnc_id": 17284,
    "hgnc_symbol": "POT1",
    "interval_id": null,
    "mean_coverage": 28.279292186937408,
    "start": 124462440
  }
]
```

File: src/chanjo2/meta/handle_d4.py [hunk 1]
```diff
@@ -48,31 +48,38 @@ def intervals_coverage(
...
+    """Use NumPy to calculate the coverage completeness over a list of threshold values for a chromosomal region."""
 
     if completeness_threholds is None:
         return []
 
-    region_size: int = stop - start
-    per_base_depth: ndarray = d4_file.load_to_np(f"{chromosome}:{start}-{stop}")
-
     completeness_values: List[Tuple[int, Decimal]] = []
+    total_region_length = sum([interval[2] - interval[1] for interval in intervals])
```
Comment (henrikstranneheim) [inline src/chanjo2/meta/handle_d4.py:62]:
```suggestion
    total_region_length: int = sum([interval[2] - interval[1] for interval in intervals])
```
I wonder if one could do something more readable like:
```
 total_region_length: int = sum([interval[CHR_STOP] - interval[CHR_START] for interval in intervals])
```

File: src/chanjo2/meta/handle_d4.py [hunk 2]
```diff
@@ -118,16 +126,15 @@ def get_genes_transcript_coverage(
             ensembl_gene_ids=[gene.ensembl_id],
             limit=None,
         )
+        intervals: List[Tuple[str, int, int]] = [
+            (transcript.chromosome, transcript.start, transcript.stop)
+            for transcript in gene_transcripts
+        ]
```
Comment (henrikstranneheim) [inline src/chanjo2/meta/handle_d4.py:132]: Could this made into a general get_intervals function?

File: src/chanjo2/meta/handle_d4.py [hunk 3]
```diff
@@ -118,16 +126,15 @@ def get_genes_transcript_coverage(
...
         if gene_transcripts:
             mean_gene_cov: float = mean(
-                intervals_mean_coverage(
-                    d4_file=d4_file,
-                    intervals=[
-                        (transcript.chromosome, transcript.start, transcript.stop)
-                        for transcript in gene_transcripts
-                    ],
-                )
+                intervals_mean_coverage(d4_file=d4_file, intervals=intervals)
```
Comment (henrikstranneheim) [inline src/chanjo2/meta/handle_d4.py:137]: Is this a get method?

File: src/chanjo2/meta/handle_d4.py [hunk 4]
```diff
@@ -139,6 +146,11 @@ def get_genes_transcript_coverage(
                 start=gene.start,
                 end=gene.stop,
                 mean_coverage=mean_gene_cov,
+                completeness=evaluate_intervals_completeness(
```
Comment (henrikstranneheim) [inline src/chanjo2/meta/handle_d4.py:149]: Evaluate or calculate or get? It seems to return completeness (no matter what).

**Review by henrikstranneheim — APPROVED** [review]
Nice 👍

File: src/chanjo2/meta/handle_d4.py (hunk 3 above)
Comment (northwestwitch) [reply src/chanjo2/meta/handle_d4.py]: Yes. renamed!

File: src/chanjo2/meta/handle_d4.py (hunk 2 above)
Comment (northwestwitch) [reply src/chanjo2/meta/handle_d4.py]: Yes, why not. Fixed!

File: src/chanjo2/meta/handle_d4.py (hunk 4 above)
Comment (northwestwitch) [reply src/chanjo2/meta/handle_d4.py]: Renamed to get, to be consistent with the get_intervals_mean_coverage function. Thanks!

**Pull request merged**